From 9b507a5d3a9164d6647d7bbb579014f2aa80249c Mon Sep 17 00:00:00 2001
From: Cyril Hrubis <chrubis@suse.cz>
Date: Fri, 21 Oct 2011 18:30:56 +0200
Subject: [PATCH 2/3] Fix sigaction/16-1.c

* Use return value from pthread functions rather than errno
* Fix typo in error propagation

Signed-off-by: Cyril Hrubis <chrubis@suse.cz>
---
 .../conformance/interfaces/sigaction/16-1.c        |   16 ++++++++--------
 1 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/testcases/open_posix_testsuite/conformance/interfaces/sigaction/16-1.c b/testcases/open_posix_testsuite/conformance/interfaces/sigaction/16-1.c
index fe8c006..19f1c6c 100644
--- a/testcases/open_posix_testsuite/conformance/interfaces/sigaction/16-1.c
+++ b/testcases/open_posix_testsuite/conformance/interfaces/sigaction/16-1.c
@@ -159,8 +159,8 @@ int test_sig(struct sig_info *s)
 	ready = 0;
 
 	label = "pthread_create()";
-	rc = pthread_create(&child, NULL, threaded, s);
-	if (rc)
+	errno = pthread_create(&child, NULL, threaded, s);
+	if (errno)
 		goto done;
 
 	/*
@@ -175,8 +175,8 @@ int test_sig(struct sig_info *s)
 	sched_yield();
 
 	label = "pthread_kill()";
-	rc = pthread_kill(child, s->sig);
-	if (rc)
+	errno = pthread_kill(child, s->sig);
+	if (errno)
 		goto done;
 
 	while (!s->caught)
@@ -188,14 +188,14 @@ int test_sig(struct sig_info *s)
 		goto done;
 
 	label = "pthread_join()";
-	rc = pthread_join(child, &thread_status);
-	if (rc)
+	errno = pthread_join(child, &thread_status);
+	if (errno)
 		goto done;
 
 	sem_destroy(&sem);
 
-	status = ((long) thread_status) && 0xFFFFFFFF;
-
+	status = ((long) thread_status) & 0xFFFFFFFF;
+	
 	return status;
 
 done:
-- 
1.7.3.4

